components/core/src/clp/Grep.cpp

```diff
@@ -540,76 +557,346 @@ std::optional<Query> Grep::process_raw_query(
-                        {}
-                };
-            case SubQueryMatchabilityResult::MayMatch:
-                sub_queries.push_back(std::move(sub_query));
-                break;
-            case SubQueryMatchabilityResult::WontMatch:
-            default:
-                // Do nothing
-                break;
+            // Update combination of ambiguous tokens
+            type_of_one_token_changed = false;
+            for (auto* ambiguous_token : ambiguous_tokens) {
+                if (ambiguous_token->change_to_next_possible_type()) {
+                    type_of_one_token_changed = true;
+                    break;
+                }
+            }
+        }
+    } else {
+        // DFA search
+        static vector<set<QueryLogtype>> query_matrix(processed_search_string.size());
+        static bool query_matrix_set = false;
+        for (uint32_t i = 0; i < processed_search_string.size() && false == query_matrix_set; i++) {
+            for (uint32_t j = 0; j <= i; j++) {
+                std::string current_string = processed_search_string.substr(j, i - j + 1);
+                std::vector<QueryLogtype> suffixes;
+                clp::SearchToken search_token;
+                if (current_string == "*") {
+                    suffixes.emplace_back('*', "*", false);
+                } else {
+                    // Add * if preceding and proceeding characters are *
+                    bool prev_star = j > 0 && processed_search_string[j - 1] == '*';
+                    bool next_star = i < processed_search_string.back() - 1
+                                     && processed_search_string[i + 1] == '*';
+                    if (prev_star) {
+                        current_string.insert(0, "*");
+                    }
+                    if (next_star) {
+                        current_string.push_back('*');
+                    }
+                    bool is_surrounded_by_delims = false;
+                    if ((j == 0 || current_string[0] == '*'
+                         || forward_lexer.is_delimiter(processed_search_string[j - 1]))
+                        && (i == processed_search_string.size() - 1 || current_string.back() == '*'
+                            || forward_lexer.is_delimiter(processed_search_string[i + 1])))
+                    {
+                        is_surrounded_by_delims = true;
+                    }
+                    bool contains_wildcard = false;
+                    set<uint32_t> schema_types;
+                    // All variables must be surrounded by delimiters
+                    if (is_surrounded_by_delims) {
+                        log_surgeon::ParserInputBuffer parser_input_buffer;
+                        std::string regex_search_string;
+                        bool contains_central_wildcard = false;
+                        uint32_t pos = 0;
+                        for (char const& c : current_string) {
+                            if (c == '*') {
```

Comment:
* Does this handle escaped wildcards in the original query?
* Am I missing something or are we not handling `?`?

Reply:
Both escaped characters and '?' wildcard are handled now in schema search code, but subqueries can't handle '?' wildcard so it currently just replaces '?' with '*'. Once subqueries can handle '?' then no modifications should be needed to be made to schema search code.